### ISSUE TITLE
Improve reply upload method

### ISSUE THREAD

**Issue by p2004a** [beyond-all-reason/ansible-spads-setup#19]
Currently reply upload is just a single call to copy file over sftp. If that fails, it fails, and is not retried in any way. We also don't have anything to monitor that.

- [ ] Improve replay upload to instead of instant upload, try to queue it and execute in the backgroud with retries
- [ ] Add monitoring for the uploads so we know when upload is broken in some server

**Comment by StanczakDominik** [comment]
I assume this is https://github.com/beyond-all-reason/ansible-spads-setup/blob/6020885c41593b949174096ed5f0bdb66868e796/group_vars/all/vars.yaml#L72

> spads_endGameCommandPath: "{{ spads_install_path }}/upload_replay.sh"

Could you help me find that script? :)

**Comment by StanczakDominik** [comment]
Nevermind, found it in https://github.com/beyond-all-reason/spads_config_bar/blob/main/upload_replay.sh ! Would a rewrite in Python be okay here?

**Comment by p2004a** [comment]
Yes, Python is totally ok, I would do it in Python myself :), many people know it and it's super easy to integrate when depending on libraries packaged in os.

From the implementation side, I would expect that the piece called by SPADs to upload is quick and effectively in practice is impossible to fail unless there is hardware outage or something like that. Basically it should only queue upload fully locally on the server (add symlink in some special upload queue directory or something like that), and then there is a central service that does all the uploading, retrying, monitoring etc.

**Comment by p2004a** [comment]
Also: replay upload is idempotent, you can upload the same file twice, and it will be handled correctly by the replay server, deduplicated by game id.

**Comment by p2004a** [comment]
Another idea: I've learned of existence of https://www.freedesktop.org/software/systemd/man/systemd.path.html and I plan to use it for doing more event, and not only time driven actuations of different syncing pieces. Might be useful also here if the upload daemon was not implemented as a continuously running job.
